akka-http-core/src/test/scala/akka/http/impl/engine/client/TlsEndpointVerificationSpec.scala

```diff
@@ -47,6 +51,12 @@ class TlsEndpointVerificationSpec extends AkkaSpec("""
         e shouldBe an[Exception]
```

Comment:
In this file `should accept certificates signed by known CA` fails now (given the `HostnameVerifier`), because:

```
[DEBUG] [12/17/2015 14:33:43.694] [TlsEndpointVerificationSpec-akka.test.stream-dispatcher-6] [akka://TlsEndpointVerificationSpec/user/StreamSupervisor-0/flow-3-3-unknown-operation] hostnameVerifier: com.typesafe.sslconfig.ssl.DefaultHostnameVerifier@10618699
session.getId = List(86, 114, -71, -73, -77, -103, 37, 74, 89, -29, 59, -127, 18, 100, 99, -95, 75, -61, -112, -91, -101, 125, 57, 118, -29, -21, 104, 88, -23, -89, -11, 39)
session.getPeerHost = null
[DEBUG] [12/17/2015 14:33:43.713] [TlsEndpointVerificationSpec-akka.test.stream-dispatcher-9] [akka://TlsEndpointVerificationSpec/user/StreamSupervisor-0/flow-3-2-unknown-operation] hostname = WAT
LOG: verify: hostname = WAT, sessionId (base64) = VnK5t7OZJUpZ4zuBEmRjoUvDkKWbfTl24+toWOmn9Sc=
LOG: Can't verify principal, no principal
LOG: verify: returning false
```

Do you think the generated cert is wrong and the test properly fails given the verifier, or the other way around?
This ties in to my question on https://github.com/akka/akka/pull/19219/files#r47908784 since you'll notice the hostInfo was `None`, so we should skip verifying?